### PR TITLE
Small UX updates to comments

### PR DESCRIPTION
**Pull request by cadeParade**
## Description
- Sort snapshot list to display single snapshots with comments before groups with no comments.
- Blur focus of reply and new thread input after using keyboard to submit or cancel comment
- When using keyboard to confirm or cancel approval dialog, stop propagation of events so nothing happens to the underlying components.

### CODE REVIEW

File: app/components/snapshot-list.js [hunk 1]
```diff
@@ -33,27 +33,56 @@ export default Component.extend(EKMixin, {
...
+    false,
+  ),
+
   _approvedSingleSnapshots: filterBy('_singleSnapshotsChanged', 'isApproved', true),
 
   _groupedSnapshotsChanged: readOnly('_snapshotGroups.groups'),
 
   numSnapshotsUnchanged: computed('build.totalSnapshots', 'snapshotsChanged.length', function() {
-    return get(this, 'build.totalSnapshots') - get(this, 'snapshotsChanged.length');
+    return this.build.totalSnapshots - this.snapshotsChanged.length;
```
Comment (cadeParade) [inline app/components/snapshot-list.js]: Many of the changes to this file are converting old syntax -- `this.get('foo')` -- to new syntax -- `this.foo`.

Comment (cadeParade) [reply app/components/snapshot-list.js]: The rest are updating the sort order.

File: tests/integration/components/snapshot-list-test.js [hunk 2]
```diff
@@ -224,65 +224,161 @@ describe('Integration: SnapshotList', function() {
...
-        expect(SnapshotList.snapshotBlocks[4].isApproved).to.equal(true);
-        await percySnapshot(this.test);
-      });
+      this.setProperties({snapshotsChanged, build, stub, browser, numSnapshotsUnchanged: 0});
+
+      await this.render(hbs`{{snapshot-list
+        snapshotsChanged=snapshotsChanged
+        snapshotsUnchanged=snapshotsUnchanged
+        build=build
+        createReview=stub
```
Comment (unicornzero) [inline tests/integration/components/snapshot-list-test.js]: ooooh, the `stub` on the fly here seems handy and readable!

File: tests/integration/components/snapshot-list-test.js [hunk 3]
```diff
@@ -224,65 +224,161 @@ describe('Integration: SnapshotList', function() {
...
+
+      const seventhBlock = SnapshotList.snapshotBlocks[6];
+      expectIsGroup(seventhBlock);
+      expectGroupSnapshotCount(seventhBlock, '3');
+
+      const eighthBlock = SnapshotList.snapshotBlocks[7];
+      expectIsSnapshot(eighthBlock);
+      expectSnapshotName(eighthBlock, approvedSingleSnapshotsWithoutCommentsTitle);
+
+      await percySnapshot(this.test);
```
Comment (unicornzero) [inline tests/integration/components/snapshot-list-test.js]: Wow, this is one comprehensive test!!!

File: app/components/snapshot-list.js [hunk 4]
```diff
@@ -62,10 +91,14 @@ export default Component.extend(EKMixin, {
   // want snapshots to change order until the page is refreshed.
   // Unless the browser changes, then we want it to be in the new order for the browser.
   snapshotBlocks: computed('activeBrowser', function() {
-    return get(this, '_unapprovedGroups').concat(
-      get(this, '_unapprovedSingleSnapshots'),
-      get(this, '_approvedGroups'),
-      get(this, '_approvedSingleSnapshots'),
+    return this._unapprovedGroupsWithComments.concat(
```
Comment (unicornzero) [inline app/components/snapshot-list.js]: nice and clean!

**Pull request merged**